src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs

```diff
@@ -96,49 +105,81 @@ private void WriteParameter(CodeWriter writer, ServiceClientParameter clientPara
-                writer.Line($"request.Uri.Reset(new {writer.Type(typeof(Uri))}($\"{urlText}\"));");
+                writer.Line($"request.Method = {typeof(RequestMethod)}.{method.ToRequestMethodName()};");
 
-                foreach (var segment in operation.Request.PathSegments)
+                var uriType = writer.Type(typeof(Uri));
+                using (nextPage ? writer.If($"{uriType}.IsWellFormedUriString(nextLinkUrl, UriKind.Absolute)") : default)
                 {
-                    WritePathSegment(writer, segment);
+                    if (nextPage)
+                    {
+                        writer.Line($"request.Uri.Reset(new {uriType}(nextLinkUrl));");
+                    }
                 }
-
-                foreach (var header in operation.Request.Headers)
+                using (nextPage ? writer.Else() : default)
                 {
-                    WriteHeader(writer, header);
+                    //TODO: Add logic to escape the strings when specified, using Uri.EscapeDataString(value);
+                    //TODO: Need proper logic to convert the values to strings. Right now, everything is just using default ToString().
+                    //TODO: Need logic to trim duplicate slashes (/) so when combined, you don't end  up with multiple // together
+                    var urlText = String.Join(String.Empty, operation.Request.HostSegments.Select(s => s.IsConstant ? s.Constant.Value : "{" + s.Parameter.Name + "}"));
+                    var nextLinkText = nextPage ? "{nextLinkUrl}" : string.Empty;
+                    writer.Line($"request.Uri.Reset(new {uriType}($\"{urlText}{nextLinkText}\"));");
+
+                    if (!nextPage)
+                    {
+                        foreach (var segment in operation.Request.PathSegments)
+                        {
+                            WritePathSegment(writer, segment);
+                        }
+
+                        //TODO: Duplicate code between query and header parameter processing logic
+                        foreach (var queryParameter in operation.Request.Query)
+                        {
+                            WriteQueryParameter(writer, queryParameter);
+                        }
+                    }
                 }
 
-                //TODO: Duplicate code between query and header parameter processing logic
-                foreach (var queryParameter in operation.Request.Query)
+                foreach (var header in operation.Request.Headers)
                 {
-                    WriteQueryParameter(writer, queryParameter);
+                    WriteHeader(writer, header);
                 }
 
-                if (operation.Request.Body is ObjectRequestBody body && body.Serialization is JsonSerialization jsonSerialization)
+                if (!nextPage && operation.Request.Body is ObjectRequestBody body && body.Serialization is JsonSerialization jsonSerialization)
```

Comment:
Having `nextPage` everywhere is very confusing.

Reply:
I agree.